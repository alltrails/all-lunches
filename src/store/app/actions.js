import { createAsyncAction, createAsyncTypes } from 'lib/reduxUtils';
// Async
export const INITIALIZE_APPLICATION = createAsyncTypes('app/initializeApplication');
export const initializeApplication = createAsyncAction(INITIALIZE_APPLICATION);
