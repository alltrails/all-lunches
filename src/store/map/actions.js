import { createAsyncAction, createAsyncTypes } from 'lib/reduxUtils';

// Async
export const ADD_REPORT = createAsyncTypes('map/addReport');
export const addReport = createAsyncAction(ADD_REPORT);

export const QUERY_AREA = createAsyncTypes('map/queryArea');
export const queryArea = createAsyncAction(QUERY_AREA);
