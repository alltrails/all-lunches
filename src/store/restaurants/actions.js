import { createAction, createAsyncAction, createAsyncTypes } from 'lib/reduxUtils';

// Sync
export const SET_HIGHLIGHTED_RESTAURANT_ID = 'restaurants/setHighlightedRestaurantId';
export const setHighlightedRestaurantId = createAction(SET_HIGHLIGHTED_RESTAURANT_ID);

export const SET_FILTER_OPTION = 'restaurants/setFilterOption';
export const setFilterOption = createAction(SET_FILTER_OPTION);

// Async
export const QUERY_AREA = createAsyncTypes('restaurants/queryArea');
export const queryArea = createAsyncAction(QUERY_AREA);

export const UPDATE_FAVORITED_RESTAURANTS = createAsyncTypes(
  'restaurants/updateFavoritedRestaurants',
);
export const updateFavoritedRestaurants = createAsyncAction(UPDATE_FAVORITED_RESTAURANTS);

export const FETCH_FAVORITED_RESTAURANTS = createAsyncTypes(
  'restaurants/fetchFavoritedRestaurants',
);
export const fetchFavoritedRestaurants = createAsyncAction(FETCH_FAVORITED_RESTAURANTS);
