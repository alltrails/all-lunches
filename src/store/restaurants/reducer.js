/* eslint-disable default-param-last */
import {
  UPDATE_FAVORITED_RESTAURANTS,
  FETCH_FAVORITED_RESTAURANTS,
  QUERY_AREA,
  SET_FILTER_OPTION,
  SET_SELECTED_RESTAURANT_ID,
} from './actions';

export const INITIAL_STATE = {
  favoritedItemIds: [],
  highlightedRestaurantId: null,
  isLoading: false,
  isUpdatingFavorites: false,
  restaurants: null,
  selectedFilterOption: null,
};

export default (state = INITIAL_STATE, action) => {
  const { payload, type } = action;

  switch (type) {
    case SET_FILTER_OPTION:
      return {
        ...state,
        selectedFilterOption: payload,
      };
    case SET_SELECTED_RESTAURANT_ID:
      return {
        ...state,
        highlightedRestaurantId: payload,
      };
    case UPDATE_FAVORITED_RESTAURANTS.PENDING:
      return {
        ...state,
        isUpdatingFavorites: true,
      };
    case UPDATE_FAVORITED_RESTAURANTS.ERROR:
      return {
        ...state,
        isUpdatingFavorites: false,
      };
    case UPDATE_FAVORITED_RESTAURANTS.SUCCESS:
      return {
        ...state,
        favoritedItemIds: payload,
        isUpdatingFavorites: false,
      };
    case FETCH_FAVORITED_RESTAURANTS.SUCCESS:
      return {
        ...state,
        favoritedItemIds: payload,
      };
    case QUERY_AREA.PENDING:
      return {
        ...state,
        isLoading: true,
      };
    case QUERY_AREA.ERROR:
      return {
        ...state,
        isLoading: false,
      };
    case QUERY_AREA.SUCCESS:
      return {
        ...state,
        isLoading: false,
        restaurants: payload,
      };
    default:
      return state;
  }
};
