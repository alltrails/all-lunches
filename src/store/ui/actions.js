import { createAction } from 'lib/reduxUtils';

// Sync
export const TOGGLE_OPEN_TOAST = 'ui/toggleOpenToast';
export const toggleOpenToast = createAction(TOGGLE_OPEN_TOAST);
