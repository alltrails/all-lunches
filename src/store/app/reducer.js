/* eslint-disable @typescript-eslint/default-param-last */
import { INITIALIZE_APPLICATION } from './actions';

export const INITIAL_STATE = {
  error: null,
  isLoading: false,
};

export default (state = INITIAL_STATE, action) => {
  const { payload, type } = action;

  switch (type) {
    case INITIALIZE_APPLICATION.PENDING:
      return {
        ...state,
        error: null,
        isLoading: true,
      };
    case INITIALIZE_APPLICATION.ERROR:
      return {
        ...state,
        error: payload,
        isLoading: false,
      };
    case INITIALIZE_APPLICATION.SUCCESS:
      return {
        ...state,
        isLoading: false,
      };
    default:
      return state;
  }
};
