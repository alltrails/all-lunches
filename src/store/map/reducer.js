/* eslint-disable default-param-last */
import { mergeReports } from 'lib/mapUtils';
import { ADD_REPORT, QUERY_AREA } from './actions';

export const INITIAL_STATE = {
  isLoading: false,
  reports: null,
  numOfReports: 0,
};

export default (state = INITIAL_STATE, action) => {
  const { payload, type } = action;

  switch (type) {
    case ADD_REPORT.PENDING:
      return {
        ...state,
        isLoading: true,
      };
    case ADD_REPORT.ERROR:
      return {
        ...state,
        isLoading: false,
      };
    case ADD_REPORT.SUCCESS:
      return {
        ...state,
        isLoading: false,
        reports: state.reports ? mergeReports(state.reports, payload) : payload,
        numOfReports: state.numOfReports ? state.numOfReports + 1 : 1,
      };
    case QUERY_AREA.SUCCESS: {
      const { numOfReports, reports } = payload;

      return {
        ...state,
        reports,
        numOfReports,
      };
    }

    default:
      return state;
  }
};
