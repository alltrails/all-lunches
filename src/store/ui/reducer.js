/* eslint-disable default-param-last */

import { TOGGLE_OPEN_TOAST } from 'store/ui/actions';
import { QUERY_AREA, UPDATE_FAVORITED_RESTAURANTS } from 'store/restaurants/actions';

export const INITIAL_STATE = {
  currentOpenToast: null,
};

const successToastMessages = {
  [UPDATE_FAVORITED_RESTAURANTS.SUCCESS]: {
    message: 'Successfully added your favorite restaurant!',
    type: 'success',
  },
  [QUERY_AREA.SUCCESS]: { message: 'Successfully fetched restaurants!', type: 'success' },
};

export default (state = INITIAL_STATE, action) => {
  const { payload, type } = action;

  switch (type) {
    case TOGGLE_OPEN_TOAST:
      return {
        ...state,
        currentOpenToast: payload,
      };
    case UPDATE_FAVORITED_RESTAURANTS.PENDING:
    case QUERY_AREA.PENDING:
      return {
        ...state,
        currentOpenToast: null,
      };
    case UPDATE_FAVORITED_RESTAURANTS.SUCCESS:
    case QUERY_AREA.SUCCESS:
      return {
        ...state,
        currentOpenToast: successToastMessages[type],
      };
    case UPDATE_FAVORITED_RESTAURANTS.ERROR:
    case QUERY_AREA.ERROR:
      return {
        ...state,
        currentOpenToast: { message: payload, type: 'error' },
      };
    default:
      return state;
  }
};
