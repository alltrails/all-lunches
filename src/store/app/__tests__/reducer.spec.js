/* eslint-env jest */
import * as types from '../actions';
import reducer, { INITIAL_STATE } from '../reducer';

describe('App reducer', () => {
  const isLoadingState = {
    ...INITIAL_STATE,
    isLoading: true,
  };

  it('returns the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  describe('INITIALIZE_APPLICATION.PENDING', () => {
    it('works as expected', () => {
      expect(
        reducer(INITIAL_STATE, {
          type: types.INITIALIZE_APPLICATION.PENDING,
        }),
      ).toMatchSnapshot();
    });
  });

  describe('INITIALIZE_APPLICATION.SUCCESS', () => {
    it('works as expected', () => {
      expect(
        reducer(isLoadingState, {
          type: types.INITIALIZE_APPLICATION.SUCCESS,
        }),
      ).toMatchSnapshot();
    });
  });

  describe('INITIALIZE_APPLICATION.ERROR', () => {
    it('works as expected', () => {
      expect(
        reducer(isLoadingState, {
          payload: 'looks like there was an error',
          type: types.INITIALIZE_APPLICATION.ERROR,
        }),
      ).toMatchSnapshot();
    });
  });
});
