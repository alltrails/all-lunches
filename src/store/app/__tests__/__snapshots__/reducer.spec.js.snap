// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App reducer INITIALIZE_APPLICATION.ERROR works as expected 1`] = `
Object {
  "error": "looks like there was an error",
  "isLoading": false,
}
`;

exports[`App reducer INITIALIZE_APPLICATION.PENDING works as expected 1`] = `
Object {
  "error": null,
  "isLoading": true,
}
`;

exports[`App reducer INITIALIZE_APPLICATION.SUCCESS works as expected 1`] = `
Object {
  "error": null,
  "isLoading": false,
}
`;

exports[`App reducer returns the initial state 1`] = `
Object {
  "error": null,
  "isLoading": false,
}
`;
